src/openrct2/entity/Guest.cpp

```diff
@@ -2591,6 +2592,16 @@ static void peep_update_ride_at_entrance_try_leave(Guest* peep)
     }
 }
 
+static bool peep_wants_to_use_ride_with_umbrella(Guest* guest, Ride* ride)
```

Comment:
We are moving away from functions and towards methods. Could you make this into a method of Guest?

Reply:
I would change this to something like ShouldRideWhileRaining and make it do the shelter check too.

Reply:
Thanks for the suggestions! I made the requested changes :).